src/main/java/com/endava/internship/warmup/domain/service/ArrayProcessorWithForLoops.java

```diff
@@ -0,0 +1,354 @@
+
+import java.util.function.IntPredicate;
+import java.util.function.ToIntFunction;
+
+public class ArrayProcessorWithForLoops implements ArrayProcessor {
+
+    /**
+     * Return true if there are no numbers that divide by 10
+     *
+     * @param input non-null immutable array of ints
+     */
+    @Override
+    public boolean noneMatch(final int[] input) {
+        int numberOfAppearance = 0;
+
+        for (int num : input) {
+            if (num % 10 == 0) {
+                numberOfAppearance++;
+            }
+        }
+
+        return numberOfAppearance == 0;
+    }
+
+    /**
+     * Return true if at least one value in input matches the predicate
+     *
+     * @param input     non-null immutable array of ints
+     * @param predicate invoke the predicate.test(int value) on each input element
+     */
+    @Override
+    public boolean someMatch(final int[] input, IntPredicate predicate) {
+
+
+        for (int num : input) {
+            if (predicate.test(num)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Return true if all values processed by function, matches the predicate
+     *
+     * @param input     non-null immutable array of Strings. No element is null
+     * @param function  invoke function.applyAsInt(String value) to transform all the input elements into an int value
+     * @param predicate invoke predicate.test(int value) to test the int value obtained from the function
+     */
+    @Override
+    public boolean allMatch(final String[] input,
+                            ToIntFunction<String> function,
+                            IntPredicate predicate) {
+        int nrOfMatches = 0;
+        for (String stringValue : input) {
+            int intValue = function.applyAsInt(stringValue);
+            if (predicate.test(intValue)) {
+                nrOfMatches++;
+            }
+        }
+
+        return nrOfMatches == input.length;
+    }
+
+    /**
+     * Copy values into a separate array from specific index to stopindex
+     *
+     * @param input          non-null array of ints
+     * @param startInclusive the first index of the element from input to be included in the new array
+     * @param endExclusive   the last index prior to which the elements are to be included in the new array
+     * @throws IllegalArgumentException when parameters are outside of input index bounds
+     */
+    @Override
+    public int[] copyValues(int[] input, int startInclusive, int endExclusive) throws IllegalArgumentException {
+        if (endExclusive <= startInclusive || startInclusive < 0) {
+            throw new IllegalArgumentException();
+        }
+        int newArrayLength = endExclusive - startInclusive;
+        int[] newArray = new int[newArrayLength];
+        int index = 0;
+        for (int i = startInclusive; i <= newArrayLength; i++) {
+            if (i >= startInclusive && i < endExclusive) newArray[index++] = input[i];
```

Comment:
I've seen often in a convention at Endava, to use the syntax 
if () { 
} 
for some reason, people consider that no brackets {} version makes it a bit less readable even if you execute only one statement when the if condition is true. Personally i am indifferent, but at real projects they might require of you to always use brackets {}